bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -152,63 +157,70 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
 
         HttpClient client = new HttpClient();
 
+        try {
+            client.start();
+        } catch (Exception e1) {
+            logger.warn("HttpClient can not be stared !", e1);
+        }
+
         // only configure a proxy if a host is provided
         if (StringUtils.isNotBlank(proxyHost) && proxyPort != null && shouldUseProxy(url, nonProxyHosts)) {
-            client.getHostConfiguration().setProxy(proxyHost, proxyPort);
-            if (StringUtils.isNotBlank(proxyUser)) {
-                client.getState().setProxyCredentials(AuthScope.ANY,
-                        new UsernamePasswordCredentials(proxyUser, proxyPassword));
-            }
+            ProxyConfiguration proxyConfig = client.getProxyConfiguration();
+            HttpProxy proxy = new HttpProxy(proxyHost, proxyPort);
+
+            Authentication.Result result = createBasicAuthenticationResult(proxyHost, proxyPort, proxyUser,
+                    proxyPassword, true);
+            // Adding an authentication result to the authentication store allows preemptive authentication
+            client.getAuthenticationStore().addAuthenticationResult(result);
+
+            proxyConfig.getProxies().add(proxy);
         }
 
-        HttpMethod method = HttpUtil.createHttpMethod(httpMethod, url);
-        method.getParams().setSoTimeout(timeout);
-        method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, new DefaultHttpMethodRetryHandler(3, false));
+        HttpMethod method = HttpUtil.createHttpMethod(httpMethod);
+
+        Request request = client.newRequest(url).method(method).timeout(timeout, TimeUnit.MILLISECONDS);
+        RetryListener listener = new RetryListener(request, DEFAULT_NUMBER_OF_RETRIES);
+        request.onComplete(listener);
```

Comment:
You retry logic is not working. Your listener will be informed, but your logic in this method will continue and simply return null on a first unsuccessful request. So even if requests 2, 3 or 4 are successful, the caller won't get the result.